src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1233,8 +1285,69 @@ private SourceColumnMetadata GetColumnMetadata(int ordinal)
                     method = ValueMethod.GetValue;
                 }
             }
+            else if (_rowSourceType == ValueSourceType.IDataReader)
+            {
+                isSqlType = false;
+                isDataFeed = false;
+
+                Type t = ((IDataReader)_rowSource).GetFieldType(ordinal);
```

Comment:
@Wraith2 this is the one aspect I would consider is _possibly_ a breaking change.

What if a consumer has a long-standing custom `IDataReader` implementation that throws a `NotImplementedException` for `GetFieldType`?

Is that something we even need to consider?

Reply:
Compatibility is always a major concern. On the other hand the upgrade from System to Microsoft version isn't a simple drop in and if it's worth the cost then it should be fine. It's down to @cheenamalhotra and the official MS folks, i'm just a contributor here 😁

Reply:
@cmeyertons 

I would consider that case and handle it fallback to default.

Reply:
done